pennylane/ops/mid_circuit_measure.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Mid-circuit measurements and associated operations.
+"""
+from typing import TypeVar, Generic, Type
+import uuid
+from pennylane.operation import Operation
+
+
+def mid_measure(wire):
+    """
+    Create a mid-circuit measurement and return an outcome.
+
+    .. code-block:: python
+
+        m0 = qml.mid_measure(0)
+    """
+    measurement_id = str(uuid.uuid4())[:8]
+    _MidCircuitMeasure(wire, measurement_id)
+    return MeasurementDependantValue(measurement_id, 0, 1)
+
+
+class _MidCircuitMeasure(Operation):
+    """
+    Operation to perform mid-circuit measurement.
+    """
+
+    num_wires = 1
+
+    def __init__(self, wires, measurement_id):
+        self.measurement_id = measurement_id
```

Comment:
should we maybe pass the `measurement_id` as a parameter to the base class? and then have a `measurement_id` property that will retrieve it?

Reply:
Not sure I follow. What would be the effect of that? The base class doesn't have a concept of a `measurement_id`.

Reply:
This has now been outdated because a `MeasurementProcess` is being used.